test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -37,6 +38,35 @@ func StatefulSetIsReady(mdb *mdbv1.MongoDB) func(t *testing.T) {
 	}
 }
 
+func StatefulSetHasOwnerReference(mdb *mdbv1.MongoDB) func(t *testing.T) {
+	return func(t *testing.T) {
+		sts := appsv1.StatefulSet{}
+		err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: f.Global.OperatorNamespace}, &sts)
+		if err != nil {
+			t.Fatal(fmt.Errorf("%s", err))
+		}
+
+		ownerReferenceMdb :=
+			*metav1.NewControllerRef(mdb, schema.GroupVersionKind{
+				Group:   mdbv1.SchemeGroupVersion.Group,
+				Version: mdbv1.SchemeGroupVersion.Version,
+				Kind:    mdb.Kind,
+			})
+
+		for _, ownerReference := range sts.GetOwnerReferences() {
+			if ownerReference.APIVersion == ownerReferenceMdb.APIVersion &&
+				ownerReference.Kind == "MongoDB" &&
+				ownerReference.Name == ownerReferenceMdb.Name &&
+				ownerReference.UID == ownerReferenceMdb.UID {
+				t.Logf("StatefulSet %s/%s has the correct OwnerReference!", mdb.Namespace, mdb.Name)
+				return
+			}
+		}
+
+		t.Fatal(fmt.Errorf("StatefulSet has not a correct OwnerReference"))
```

Comment:
bear in mind that this will be the output we get when trying to investigate what went wrong, we should try and include the actual and expected values to get better insight.

Reply:
Related to the comment above: since `getOwnerReferences()` returns a slice, we would have to include every ownerReference returned? In this case we only have one but we might potentially have more